flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/MailboxOperatorTest.java

```diff
@@ -0,0 +1,146 @@
+			OneInputStreamTask::new,
+			BasicTypeInfo.INT_TYPE_INFO,
+			BasicTypeInfo.INT_TYPE_INFO);
+
+		testHarness.setupOperatorChain(new OperatorID(), new ReplicatingMailOperatorFactory())
+			.chain(new OperatorID(), new ReplicatingMailOperatorFactory(), IntSerializer.INSTANCE)
+			.finish();
+
+		testHarness.invoke();
+		testHarness.waitForTaskRunning();
+
+		testHarness.processElement(new StreamRecord<>(0));
+		testHarness.processElement(new StreamRecord<>(0));
+		testHarness.processElement(new StreamRecord<>(0));
+
+		testHarness.endInput();
+		testHarness.waitForTaskCompletion();
+
+		// with each input two mails should be processed, one of each operator in the chain
+		List<Integer> numMailsProcessed = testHarness.getOutput().stream()
+			.map(element -> ((StreamRecord<Integer>) element).getValue())
+			.collect(Collectors.toList());
+		assertThat(numMailsProcessed, is(Arrays.asList(0, 2, 4)));
+	}
+
+	private static class ReplicatingMailOperatorFactory implements OneInputStreamOperatorFactory<Integer, Integer>,
+			YieldingOperatorFactory<Integer> {
+		private MailboxExecutor mailboxExecutor;
+
+		@Override
+		public void setMailboxExecutor(MailboxExecutor mailboxExecutor) {
+			this.mailboxExecutor = mailboxExecutor;
+		}
+
+		@Override
+		public <Operator extends StreamOperator<Integer>> Operator createStreamOperator(
+				StreamTask<?, ?> containingTask,
+				StreamConfig config,
+				Output<StreamRecord<Integer>> output) {
+			ReplicatingMailOperator operator = new ReplicatingMailOperator(mailboxExecutor);
+			operator.setup(containingTask, config, output);
+			return (Operator) operator;
+		}
+
+		@Override
+		public void setChainingStrategy(ChainingStrategy strategy) {
+		}
+
+		@Override
+		public ChainingStrategy getChainingStrategy() {
+			return ChainingStrategy.ALWAYS;
+		}
+
+		@Override
+		public Class<? extends StreamOperator> getStreamOperatorClass(ClassLoader classLoader) {
+			return ReplicatingMailOperator.class;
+		}
+	}
+
+	private static class ReplicatingMailOperator extends AbstractStreamOperator<Integer>
+			implements OneInputStreamOperator<Integer, Integer> {
+
+		private final AtomicInteger mailIndex = new AtomicInteger();
+		private final MailboxExecutor mailboxExecutor;
+
+		ReplicatingMailOperator(final MailboxExecutor mailboxExecutor) {
+			this.mailboxExecutor = mailboxExecutor;
+		}
+
+		@Override
+		public void processElement(StreamRecord<Integer> element) throws Exception {
+			// for the very first element, enqueue one mail that replicates itself
+			if (mailIndex.get() == 0) {
+				mailboxExecutor.execute(new Runnable() {
+					@Override
+					public void run() {
+						try {
+							mailboxExecutor.execute(this, "Blocking mail" + mailIndex.incrementAndGet());
+						} catch (RejectedExecutionException e) {
+							// during shutdown the executor will reject new mails, which is fine for us.
+						}
+					}
+				}, "Blocking mail" + mailIndex.get());
```

Comment:
nitty nit: (here and in `testAvoidStarvation `) I think this  violates our [code style](https://flink.apache.org/contributing/code-style-and-quality-formatting.html#breaking-the-lines-of-too-long-statements). I think it should be:
```
		mailboxExecutor.execute(
			new Runnable() {
				@Override
				public void run() {
					try {
						mailboxExecutor.execute(this, "Blocking mail" + mailIndex.incrementAndGet());
					} catch (RejectedExecutionException e) {
						// during shutdown the executor will reject new mails, which is fine for us.
					}
				}
			},
			"Blocking mail" + mailIndex.get());
```

Alternatively, name the function instead of runnable? Or add a util function to ignore some specified type of the exception in `org.apache.flink.util.function.FunctionUtils`?

Reply:
Solved it by actually factoring a class `ReplicatingMail` out. Imho makes the test more readable.